### PR TITLE
[cmake] support shared or static builds

### LINKED ISSUE #69: how to compile dynamic CCTAG_LIBRARY with cuda?
I need to dynamic invocat the cctag library as a '.so' library in Python code. 
Why the code must be static if use cuda? Could you tell me how to change these?

### PR DESCRIPTION
**Pull request by fabiencastan**
Fix #69.

### CODE REVIEW

**Comment by fabiencastan** [comment]
@bubulv Can you test this fix?
It works for me on Fedora with cuda-7.0.

File: CMakeLists.txt [hunk 1]
```diff
@@ -108,6 +107,11 @@ if(WITH_CUDA)
   endif()
 
   set(CUDA_NVCC_FLAGS         "${CUDA_NVCC_FLAGS};--default-stream;per-thread")
+
+  if(CCTAG_USE_POSITION_INDEPENDENT_CODE)
```
Comment (simogasp) [inline CMakeLists.txt]: u can just check `BUILD_SHARED_LIBS` instead of using another variable

Comment (fabiencastan) [reply CMakeLists.txt]: Yes, that's the idea of removing `COMPILE_STATIC_CCTAG_LIBRARY` but position independent is a bit different.
I'm currently using the static build with position independent code to link [AliceVision](https://github.com/alicevision/AliceVision).

**Comment by bubulv** [comment]
It's Ok build shared library as you say. But I have a new question. Now I need to deel with many different size frames with cuda , but line 752 in Detection.cpp make error.
 std::cerr << "We cannot change the input frame resolution (yet)" << std::endl;

It's no problem without cuda, why we need to check 'width  != pipe1->getWidth(0)'? How it can be fix?

**Comment by bubulv** [comment]
Thank you very much. It's ok!!!!

**Pull request merged**